src/cephadm/cephadm

```diff
@@ -3450,26 +3453,159 @@ def command_list_networks():
 
 ##################################
 
-
 def command_ls():
     # type: () -> None
     ls = list_daemons(detail=not args.no_detail,
-                      legacy_dir=args.legacy_dir)
+                      legacy_dir=args.legacy_dir,
+                      target_fsid=args.fsid)
     print(json.dumps(ls, indent=4))
 
 
-def list_daemons(detail=True, legacy_dir=None):
-    # type: (bool, Optional[str]) -> List[Dict[str, str]]
-    host_version = None
-    ls = []
+def get_enabled_ceph_units(fsid: str) -> List[str]:
+    """Use the filesystem itself to determine enabled state
+
+    Enabled units are present in the target.wants directory, so by scanning 
+    this directory we can quickly determine state for all units, avoiding
+    iterating over multiple systemctl calls
+    """
+    try:
+        units = os.listdir(f'/etc/systemd/system/ceph-{fsid}.target.wants')
+    except FileNotFoundError:
+        return []
+    else:
+        return units
+
+
+def get_unit_states(fsid: str) -> Dict[str, Dict[str, str]]:
+    """Determine all systemd unit states for a given fsid
+
+    Instead of iterating over systemctl calls, one per unit, we issue a 
+    single systemctl command to return all units that match our fsid. The
+    reulting output is then parsed to provide a lookup of unit -> systemd state
+    """
+    cmd = f'systemctl --all --no-legend -t service list-units ceph-{fsid}*'
```

Comment:
```suggestion
    cmd = f'systemctl --all --no-pager -t service list-units ceph-{fsid}*'
```

This avoids the legends and doesn't pipe the output into a pager

Reply:
ACK